components/function-controller/design/extend_function_spec_proposal.md

```diff
@@ -0,0 +1,392 @@
+# Extending CRD spec for Serverless v1alpha2
+
+## Summary
+
+The current Serverless API allows for limited configuration of the generated Function's deployment. 
+Currently users can only use ENVs to pass 3rd party service credentials but it doesn't allow for volume mounted secrets (which becomes the industry standard for [service bindings](https://servicebinding.io/application-developer/)).
+Moreover users have no control over the annotations that are applied on the function runtime pod. This excludes functions pods from features enabled by annotations (i.e custom log parsers via `fluentbit.io/parser: my-regex-parser`).
+
+
+
+## Motivation
+
+Give Serverless users the ability to:
+- Configure volume mounted secrets (and config maps) for Function's subresources.
+- Configure labels and annotations for the  Function runtime pod.
+
+### Goals
+
+- Add more flexibility to the Serverless API - enable volume mounted secrets and pod annotations.
+- Organise spec attributes belonging to runtime and build-time configuration (?)
+- Provide a rollout plan to implement this functionality in small increments and avoid the need to roll out a new API version
+
+### Non-Goals
+- Define a new API version specification
```

Comment:
I don't think this would be possible. All the proposals have big changes to the current API, not just additions. I think if we are going to keep this non-goal in mind, we should include a high-level implementation plan on how to achieve that.